### ISSUE TITLE
pasystray menu not accessible via screen reader

### ISSUE THREAD

**Issue by smlx** [christophgysin/pasystray#88]
Hi, I have a [bug report in Debian](https://bugs.debian.org/861387) that the pasystray menu is not accessible - that is, it is not usable with the Gnome screen-reader application, Orca.

The user reports that padevchooser does work correctly with the same screen-reader.

Apparently accessibility features in GTK are supposed to be built-in to the standard widgets. I have had a brief dig through the GTK documentation but I'm not familiar enough with it to offer a patch for this immediately.

Could you look into adding accessibility features to the pasystray menu?

**Comment by smlx** [comment]
Ah - I think I see the problem. The icon in the system tray does not open the menu on keyboard input (<enter> or <space>), it only responds to mouse clicks. Once opened with the mouse, the menu is indeed accessible via the screen reader.

As described in the Debian bug report, you can reproduce the problem via:

1. Start pasystray in Gnome
2. Press Ctrl-Alt-Tab and select the legacy system tray applet
3. Navigate to the pasystray icon using the arrow keys
4. Press <enter> or <space> - nothing happens.

**Comment by christophgysin** [comment]
I'm using GtkStatusIcon, which is a standard widget in GTK. From what I can tell it does not seem to support keyboard events, see https://developer.gnome.org/gtk3/stable/GtkStatusIcon.html#GtkStatusIcon.signals

Looking at the padevchooser source code, it does not seem to use GtkStatusIcon, but rather implement the systray directly using X libraries. Probably because GtkStatusIcon didn't exist when padevchooser was written.

Maybe GTK developers could comment on that. It's possible that there is a way to catch keyboard events, but I don't know how. If it is not currently possible, it will probably not get fixed upstream, since GtkStatusIcon has been deprecated.

I'm not willing to rewrite GtkStatusIcon using X libraries to support this, but if somebody else does, I'd be happy to apply a PR.

**Comment by smlx** [comment]
I see. Thanks for looking into this.

Unfortunately GtkStatusIcon is [deprecated](https://bugzilla.gnome.org/show_bug.cgi?id=734826) as of Gtk+ 3.14 (they want everyone to use StatusNotifier/AppIndicator). So I guess that this problem is solved by using pasystray's AppIndicator option. I'll test it and see how it behaves with the screen-reader.

I've avoided the AppIndicator option in pasystray until now due to the missing modifier key support. But maybe that will be the only option going forward..

**Comment by smlx** [comment]
I've tested using a screen reader and this is fixed by switching to AppIndicator. Thanks for your help!
